demo-terraform/aws/simple-ec2-with-s3-access/ec2.tf

```diff
@@ -24,76 +24,82 @@ locals {
 
 ## Search for an Ubuntu AMI in your region 
 data "aws_ami" "ubuntu" {
-    most_recent = true
-    name_regex = "^.*ubuntu-focal-20.04-amd64-server-.*"    
-    owners = [ "099720109477" ] ## Canonical
-
-    filter {
-        name   = "root-device-type"
-        values = ["ebs"]
-    }
-
-    filter {
-        name   = "virtualization-type"
-        values = ["hvm"]
-    }
+  most_recent = true
+  name_regex  = "^.*ubuntu-focal-20.04-amd64-server-.*"
+  owners      = ["099720109477"] ## Canonical
+
+  filter {
+    name   = "root-device-type"
+    values = ["ebs"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
 }
 
 
 
 resource "aws_instance" "web" {
-    
-    ami = data.aws_ami.ubuntu.id
-    instance_type = var.instance_type
-
-    root_block_device {
-        encrypted = false
-        volume_size = 20
-        volume_type = "standard"
-        tags = local.default_tags
-    }
 
-    subnet_id = random_shuffle.web_subnet.result[0]
+  ami           = data.aws_ami.ubuntu.id
+  instance_type = var.instance_type
+
+  root_block_device {
+    encrypted   = false
+    volume_size = 20
+    volume_type = "standard"
+    tags        = local.default_tags
+  }
 
-    tags = merge(local.default_tags,
-        {
-            Name = "${replace(local.user_mail,"@","-")}-${random_string.bucket_suffix.result}"
-        }
-    )
+  subnet_id = random_shuffle.web_subnet.result[0]
 
-    # key_name = aws_key_pair.kp.key_name
+  tags = merge(local.default_tags,
+    {
+      Name = "${replace(local.user_mail, "@", "-")}-${random_string.bucket_suffix.result}"
+    }
+  )
+
+  # key_name = aws_key_pair.kp.key_name
+
+  vpc_security_group_ids = [aws_security_group.web_sg.id]
+  monitoring             = true
 
-    vpc_security_group_ids = [aws_security_group.web_sg.id]
+  metadata_options {
+    http_tokens   = "required"
+    http_endpoint = "disabled"
+  }
 }
 
 resource "random_shuffle" "web_subnet" {
-    input        = jsondecode(data.aws_ssm_parameter.demo_vpc_private_subnets.value)
-    result_count = 1
+  input        = jsondecode(data.aws_ssm_parameter.demo_vpc_private_subnets.value)
+  result_count = 1
 }
 
 resource "aws_security_group" "web_sg" {
-  name        = "${replace(local.user_mail,"@","-")}-${random_string.bucket_suffix.result}"
-  vpc_id      = data.aws_ssm_parameter.demo_vpc_id.value
-
-    ingress {
-        description      = "Incoming SSH"
-        from_port        = 22
-        to_port          = 22
-        protocol         = "tcp"
-        cidr_blocks      = [ "0.0.0.0/0" ]
+  name   = "${replace(local.user_mail, "@", "-")}-${random_string.bucket_suffix.result}"
+  vpc_id = data.aws_ssm_parameter.demo_vpc_id.value
+
```

Comment:
**Accurics remediation**
 
**[HIGH]: Ensure Security Groups do not have unrestricted specific ports open - (SSH,22)**
 
**Vulnerability**
Security Groups - Unrestricted Specific Ports - SSH,22. Exposed ports may allow attackers to access the application listening on that port.
 
**Remediation**
In AWS Console - 
 1. Sign in the AWS Console and go to the Amazon VPC console.
 2. In the navigation pane, select Security Groups.
 3. For each security group, perform the following actions:
      a) Select the security group.
      b) Select the Inbound Rules.
      c) Identify the rules to be removed.
      d) Remove the required rules.
 4. Click Save.

source: accurics-